common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java

```diff
@@ -291,4 +293,14 @@ private long getExpiresOn(final MicrosoftStsTokenResponse msTokenResponse) {
         return currentTimeSecs + expiresIn;
     }
 
+    private long getRefreshOn(final MicrosoftStsTokenResponse msTokenResponse) {
+        final long currentTimeMillis = System.currentTimeMillis();
+        final long currentTimeSecs = TimeUnit.MILLISECONDS.toSeconds(currentTimeMillis);
+        final long refreshIn = msTokenResponse.getRefreshIn() == null ? msTokenResponse.getExpiresIn() : msTokenResponse.getRefreshIn();
+
+        return currentTimeSecs + refreshIn;
```

Comment:
Do we need to differentiate between a case where refresh_in is set by server vs not? Current logic seems to always return refresh_in value (based on expires_in value in some cases).

Reply:
+1 on this...I think we should have some way to differentiate and I also I think the important question here is do we want clients to still do a refresh even if server doesn't return a rerfresh_in value? My initial understanding was that if the server doesn't return something here then we won't do anything apart from the (automatic) refresh that happens as part of `acquireTokenSilent` if the `AT` had expired and redeem RT to get new AT, RT, ID tokens..

Reply:
I think he's talking about in the logs.

Reply:
Yes, the question is how do we distinguish in the logs when we are debugging issues

Reply:
There is a method now AccessTokenRecord.refreshOnIsActive() which determines if server is supporting refreshIn. It's used to produce logs in LocalMSALController.